android/javatests/org/chromium/chrome/browser/BraveSwipeRefreshHandlerTest.java

```diff
@@ -0,0 +1,163 @@
+import org.chromium.chrome.browser.flags.ChromeSwitches;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.test.ChromeJUnit4ClassRunner;
+import org.chromium.third_party.android.swiperefresh.SwipeRefreshLayout;
+import org.chromium.third_party.android.swiperefresh.SwipeRefreshLayout.OnRefreshListener;
+import org.chromium.third_party.android.swiperefresh.SwipeRefreshLayout.OnResetListener;
+import org.chromium.ui.OverscrollAction;
+import org.chromium.ui.base.BackGestureEventSwipeEdge;
+import org.chromium.ui.test.util.BlankUiTestActivity;
+
+/** Tests {@link BraveSwipeRefreshHandler}. */
+@RunWith(ChromeJUnit4ClassRunner.class)
+@CommandLineFlags.Add({ChromeSwitches.DISABLE_FIRST_RUN_EXPERIENCE})
+@Batch(Batch.PER_CLASS)
+public class BraveSwipeRefreshHandlerTest {
+    private static final int ACCESSIBILITY_SWIPE_REFRESH_STRING_ID =
+            R.string.accessibility_swipe_refresh;
+
+    @ClassRule
+    public static BaseActivityTestRule<BlankUiTestActivity> activityTestRule =
+            new BaseActivityTestRule<>(BlankUiTestActivity.class);
+
+    // Can't initialize here since myActivityTestRule.getActivity() is still null.
+    private static String sAccessibilitySwipeRefreshString;
+
+    @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+    @Mock private Tab mTab;
+    private OnRefreshListener mOnRefreshListener;
+    private OnResetListener mOnResetListener;
+    private SwipeRefreshLayout mSwipeRefreshLayout;
+    private final SwipeRefreshHandler.SwipeRefreshLayoutCreator mSwipeRefreshLayoutCreator =
+            context -> {
+                mSwipeRefreshLayout = mock();
+                doAnswer((invocation) -> mOnRefreshListener = invocation.getArgument(0))
+                        .when(mSwipeRefreshLayout)
+                        .setOnRefreshListener(any());
+                doAnswer((invocation) -> mOnResetListener = invocation.getArgument(0))
+                        .when(mSwipeRefreshLayout)
+                        .setOnResetListener(any());
+                return mSwipeRefreshLayout;
+            };
+
+    @BeforeClass
+    public static void setUpSuite() {
+        activityTestRule.launchActivity(null);
+        sAccessibilitySwipeRefreshString =
+                activityTestRule.getActivity().getString(ACCESSIBILITY_SWIPE_REFRESH_STRING_ID);
+    }
+
+    @Before
+    public void setup() {
+        when(mTab.getContext()).thenReturn(activityTestRule.getActivity());
+        when(mTab.getUserDataHost()).thenReturn(new UserDataHost());
+        when(mTab.getContentView()).thenReturn(mock());
+    }
+
+    @Test
+    @SmallTest
+    public void testPullToRefreshIgnored() {
+        SwipeRefreshHandler chromeHandler =
+                SwipeRefreshHandler.from(mTab, mSwipeRefreshLayoutCreator);
+        chromeHandler.initWebContents(mock()); // Needed to enable the overscroll refresh handler.
+        BraveSwipeRefreshHandler braveHandler = BraveSwipeRefreshHandler.from(mTab);
```

Comment:
We can eliminate the code duplication by adding a helper like:
```java
private BraveSwipeRefreshHandler createBraveHandler() {
    SwipeRefreshHandler chromeHandler =
            SwipeRefreshHandler.from(mTab, mSwipeRefreshLayoutCreator);
    chromeHandler.initWebContents(mock()); // Needed to enable the overscroll refresh handler.
    return BraveSwipeRefreshHandler.from(mTab);
}
```
or by introducing a `private BraveSwipeRefreshHandler mBraveHandler;` field and moving those same lines into `setup()`.

Reply:
thanks, now is a bit cleaner, updated at https://github.com/brave/brave-core/pull/32595/commits/4fff723fe20656a973c744579b74e5ff4073012a